NetSDK/Services/Impressions/Classes/InMemoryTreatmentLog.cs

```diff
@@ -0,0 +1,77 @@
+﻿using log4net;
+using Splitio.Domain;
+using Splitio.Services.Impressions.Interfaces;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Splitio.Services.Impressions.Classes
+{
+    public class InMemoryTreatmentLog : ITreatmentLog
+    {
+        private ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>> cache;
+        private int maximumNumberOfKeysToCachePerTest;
+        private object lockObject = new Object();
+        protected static readonly ILog Logger = LogManager.GetLogger(typeof(InMemoryTreatmentLog));
+
+        public InMemoryTreatmentLog(ConcurrentDictionary<string, ConcurrentQueue<KeyImpression>> cache = null, int maximumNumberOfKeysToCachePerTest = -1)
+        {
+            this.maximumNumberOfKeysToCachePerTest = maximumNumberOfKeysToCachePerTest;
+            this.cache = cache ?? new ConcurrentDictionary<string, ConcurrentQueue<KeyImpression>>();
+        }
+
+        protected virtual void NotifyEviction(string feature, ConcurrentQueue<KeyImpression> impressions) { }
+
+
+        public ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>> FetchAllAndClear()
+        {
+            lock (lockObject)
+            {
+                var existingImpressions = new ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>>(cache);
+                cache = new ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>>();
+                return existingImpressions;
+            }
+        }
+
+        public void Log(string id, string feature, string treatment, long time)
+        {
+            if (id == null || feature == null || treatment == null || time < 0)
+            {
+                return;
+            }
+
+            lock (lockObject)
```

Comment:
We can move to Concurrent Queue approach, and then feature name in KeyImpression objects has sense, but let me know if TEST_IMPRESSIONS_REFRESH_RATE and TEST_IMPRESSIONS_CACHE_SIZE_PER_TEST are no longer neccesary. Maybe, at least, we need to configure the time that Impressions module should wait until it sends bulk impressions to api (TEST_IMPRESSIONS_REFRESH_RATE)